nodejs/expressmvc/routes/authentication-cloudhub.js

```diff
@@ -0,0 +1,139 @@
+const express = require("express");
+const { Util } = require("../util");
+const { Authentication, CompleteAuth } = require("restpki-client");
+const restpki = require("restpki-client");
+
+const uuidv4 = require("uuid/v4");
+const path = require("path");
+
+const app = express();
+
+const router = express.Router();
+const APP_ROOT = process.cwd();
+
+this.client = Util.getCloudhubClient();
+this.restPkiClient = Util.getRestPkiClient();
+
+/*
+ * GET /authentication-cloudhub
+ *
+ * This page will ask for the user identifier, so it is able to perform a request to
+ * the next page, where it will ask which cloud signature service the user wants
+ */
+router.get("/", async (req, res, next) => {
+	// Get an instance of the Authentication class.
+	const auth = new Authentication(this.restPkiClient);
+
+	// Render the authentication page.
+	res.render("authentication-cloudhub", {});
+});
+
+/*
+ * POST /pades-signature-cloudhub-restpki
+ * This is the POST request from the index page which
+ * sends the cpf and the file to be signed
+ */
+router.post("/", async (req, res, next) => {
+	try {
+		const cpf = req.body.cpf;
+		console.log(restpki.VERSION);
+
+		// remove all dots and hyphens
+		const new_cpf = cpf.replace(/[.-]/g, "");
+
+		const sessionRes = await this.client.createSessionAsync({
+			identifier: new_cpf,
+			redirectUri: `http://localhost:3000/authentication-cloudhub/session-result`,
+			type: TrustServiceSessionTypes.SingleSignature,
+		});
+
+		res.render("authentication-cloudhub/service-select", {
+			sessionRes,
+		});
+	} catch (err) {
+		next(err);
+	}
+});
+
+/*
+ * GET /authentication-cloudhub/session-result
+ * This page performs the signature process using CloudHub API session
+ * and obtains the user certificate, which is used in Pades signature
+ * process
+ */
+router.get("/session-result", async (req, res, next) => {
+	try {
+		const auth = new Authentication(this.restPkiClient);
+		const session = req.query.session;
+		const cert = await this.client.getCertificateAsync(session);
+
+		// Call the start() method, which initiates the authentication.
+		// This yields the "token", a 22-character case-sensitive URL-safe string,
+		// which represents this authentication process. We'll use this value to call
+		// the sign() method on the Web PKI component
+		// (see public/javascripts/signature-form.js) and also call the
+		// complete() method on "complete" step. This should not be
+		// mistaken with the API access token. We have encapsulated the security
+		// context choice on util.js.
+		var startResult = await auth.start(
+			Util.getSecurityContextId(res.locals.environment)
+		);
+		const token = startResult.token;
+
+		var signature = await this.client.signHashAsync({
+			hash: startResult.toSignHash,
+			digestAlgorithmOid: startResult.digestAlgorithmOid,
+			session: session,
+		});
+
+		const payload = new CompleteAuth({
+			signature: signature,
+			certificate: cert,
+		});
+
+		auth.complete(token, payload)
```

Comment:
Can you change here to use `await` instead of `.then()` method?
I believe the code will be more readable.